lib/src/Uncertainty/Bayesian/RandomWalkMetropolisHastings.cxx

```diff
@@ -98,198 +86,128 @@ RandomWalkMetropolisHastings* RandomWalkMetropolisHastings::clone() const
-      throw InvalidArgumentException(HERE) << "The initial state should have non-zero posterior proability density";
+    if (getVerbose())
+      LOGTRACE(OSS() << "rho=" << rho << " delta=" << delta_);
   }
 
-  // for each new sample
-  for (UnsignedInteger i = 0; i < size; ++ i)
-  {
-    // accumulates the updates over each component
-    Point newState(currentState_);
-
-    history_.store(currentState_);
-
-    Scalar logLikelihoodCandidate = currentPenalizedLogLikelihood_;
-
-    // update each chain component
-    for (UnsignedInteger j = 0; j < dimension; ++ j)
-    {
-      // new candidate for the j-th component
-      Point nextState(newState);
-
-      Bool nonRejectedComponent = nonRejectedComponents_.contains(j);
-      if (!nonRejectedComponent)
-      {
-        // regular MCMC
-        nextState[j] += delta[j] * proposal_[j].getRealization()[0];
-      }
-      else
-      {
-        // non-rejected component
-        nextState[j] = getPrior().getMarginal(j).getRealization()[0];
-      }
-
-      const Scalar nextPenalizedLogLikelihood = computeLogLikelihood(nextState) + getPrior().computeLogPDF(nextState);
-
-      // alpha = posterior(newstate)/posterior(oldstate)
-      const Scalar alphaLog = nextPenalizedLogLikelihood  - currentPenalizedLogLikelihood_;
-
-      // acceptance test
-      const Scalar uLog = log(RandomGenerator::Generate());
-      if (nonRejectedComponent || (uLog < alphaLog))
-      {
-        logLikelihoodCandidate = nextPenalizedLogLikelihood;
-        ++ acceptedNumber_[j];
-        ++ accepted[j];
-
-        newState[j] = nextState[j];
-      }
-    }
-
-    // reuse the penalized log-likelihood of the last accepted update
-    currentPenalizedLogLikelihood_ = logLikelihoodCandidate;
-
-    // update state
-    currentState_ = newState;
-
-    // recalibrate each component if necessary
-    if (samplesNumber_ < getBurnIn())
-    {
-      for (UnsignedInteger j = 0; j < dimension; ++ j)
-      {
-        const UnsignedInteger calibrationStep = calibrationStrategy_[j].getCalibrationStep();
-        if ((samplesNumber_ % calibrationStep) == (calibrationStep - 1))
-        {
-          // compute the current acceptation rate
-          Scalar rho = 1.0 * accepted[j] / (1.0 * calibrationStep);
-
-          // compute factor
-          Scalar factor = calibrationStrategy_[j].computeUpdateFactor(rho);
-
-          // update delta
-          delta[j] *= factor;
-
-          // reset the counter
-          accepted[j] = 0;
-
-          if (getVerbose())
-          {
-            LOGINFO( OSS() << "rho=" << rho << " factor=" << factor << " delta=" << delta.getCollection() << " accept=" << getAcceptanceRate().getCollection() );
-          }
-
-        } // calibrationStep
-      } // j
-    } // burn-in
-
-    ++ samplesNumber_;
-
-  } // for i
-
-  // Save the last state
-  history_.store(currentState_);
-  return currentState_;
+  Point prop(delta_ * instrumental_.getRealization());
```

Comment:
Could we have an accessor to the `delta_` attribute? Something like `getStepMultiplier`? That way we could inspect the result of the adaptation phase.

Reply:
`getAdaptationFactor`